packages/core/src/debugr/debugr.ts

```diff
@@ -0,0 +1,132 @@
+
+  public constructor(
+    eventDispatcher: EventDispatcher,
+    pluginManager: PluginManager<Partial<TTaskContext>, TGlobalContext>,
+    logger: Logger<Partial<TTaskContext>, TGlobalContext>,
+    plugins: Plugin<Partial<TTaskContext>, TGlobalContext>[],
+  ) {
+    this.eventDispatcher = eventDispatcher;
+    this.pluginManager = pluginManager;
+    this.logger = logger;
+
+    for (const plugin of plugins) {
+      this.pluginManager.register(plugin);
+    }
+
+    for (const logHandler of this.logger.getAllHandlers()) {
+      logHandler.injectPluginManager(this.pluginManager);
+    }
+
+    this.checkFormatters();
+  }
+
+  public static create<
+    TTaskContext extends TContextBase = TContextBase,
+    TGlobalContext extends TContextShape = {},
+  >(
+    globalContext: TGlobalContext,
+    logHandlers: LogHandler<Partial<TTaskContext>, TGlobalContext>[],
+    plugins: Plugin<Partial<TTaskContext>, TGlobalContext>[],
+  ): Debugr<Partial<TTaskContext>, TGlobalContext> {
+    const logger = new Logger<Partial<TTaskContext>, TGlobalContext>(
+      logHandlers || [],
+      globalContext,
+    );
+    const pluginManager = new PluginManager<Partial<TTaskContext>, TGlobalContext>(logger);
+
+    for (const logHandler of logHandlers) {
+      logHandler.injectPluginManager(pluginManager);
+    }
+
+    const debugr = new Debugr<Partial<TTaskContext>, TGlobalContext>(
+      new EventDispatcher(1000),
+      pluginManager,
+      logger,
+      plugins || [],
+    );
+    return debugr;
+  }
+
+  public hasPlugin(id: string): boolean {
+    return this.pluginManager.has(id);
+  }
+
+  public getPlugin<ID extends PluginId>(
+    id: ID,
+  ): Plugins<Partial<TTaskContext>, TGlobalContext>[ID] {
+    return this.pluginManager.get(id);
+  }
+
+  public hasHandler(id: string): boolean {
+    return this.logger.hasHandler(id);
+  }
+
+  public getHandler(id: string): LogHandler<Partial<TTaskContext>, TGlobalContext> | never {
+    return this.logger.getHandler(id);
+  }
+
+  public on<E extends keyof Events>(event: E, listener: Events[E]): void {
+    this.eventDispatcher.on(event, listener);
+  }
+
+  public once<E extends keyof Events>(event: E, listener: Events[E]): void {
+    this.eventDispatcher.once(event, listener);
+  }
+
+  public off<E extends keyof Events>(event: E, listener?: Events[E]): void {
+    this.eventDispatcher.off(event, listener);
+  }
+
+  public registerListeners(listeners: Partial<Events>): void {
+    this.eventDispatcher.register(listeners);
+  }
+
+  private checkFormatters(): void | never {
+    const plugins = this.pluginManager.getAll();
+    const logHandlers = this.logger.getAllHandlers();
+    const formatterPluginsErrors: string[] = [];
+    for (const logHandler of logHandlers) {
+      if (!logHandler.doesNeedFormatters) {
+        continue;
+      }
+
+      const entryFormats = new Set(...plugins.map((plugin) => plugin.entryFormat));
```

Comment:
tohle určitě může bejt mimo ten cyklus kterej projíždí log handlers, protože je to pro všechny jeho iterace to samý, a klidně by šlo ty pluginy rovnou profiltrovat přes `isFormatterPlugin`, takže to pak nebude potřeba kontrolovat znovu na L110

Reply:
oka